### ISSUE TITLE
Error when running Powershell on target machine task

### ISSUE THREAD

**Issue by sameer-kumar** [Microsoft/vsts-tasks#2302]
I'm getting below mentioned error when I am trying to execute a Powershell script on target machine from my on-prem TFS 2015 Release pipeline.

 `##[error]The term 'Set-Variable' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again. For more info please refer to http://aka.ms/powershellontargetmachinesreadme`
 
This script is a DSC configuration. I have POSH 5 RTM installed on target node.

```
Configuration Main
 {
   Node ('localhost')
   {
     WindowsFeature WebServerRole
     {
       Name = "Web-Server"
       Ensure = "Present"
     }

     WindowsFeature WebAspNet45
     {
       Name = "Web-Asp-Net45"
       Ensure = "Present"
       #Source = $Source
       DependsOn = "[WindowsFeature]WebServerRole"
     }

     WindowsFeature WebServerManagementTools
     {
        Name = "Web-Mgmt-Tools"
        Ensure = "Present"
        DependsOn = "[WindowsFeature]WebServerRole"
     }
     
     WindowsFeature SecurityWindowsAuth
     {
        Name = "Web-Windows-Auth"
        Ensure = "Present"
        DependsOn = "[WindowsFeature]WebServerRole"
     }

     WindowsFeature WcfHttpActivation
     {
        Name = "NET-WCF-HTTP-Activation45"
        Ensure = "Present"
        DependsOn = "[WindowsFeature]WebAspNet45"
     }

   }
 }

 Main
```

Am I missing something?

**Comment by sameer-kumar** [comment]
I verified that the task runs fine and executes the configuration on target node with WMF 4 (windows 2012 R2)

**Comment by kmkumaran** [comment]
@sameer-kumar, yes WMF 4 is a requirement.

**Comment by sameer-kumar** [comment]
i guess i was not able to communicate well. Let me try again. 
The above mentioned error occurs with WMF 5.0 RTM. But the configuration executes fine with POSH 4

**Comment by ShaunInIdaho** [comment]
I'm also encountering this error after upgrading to WMF 5.0.  The issue only occurs when running a script non-interactively.  It works fine on WMF 5.0 from the ISE, but when running a script as a scheduled task for example, it throws the error: "The term 'Set-Variable' is not recognized as the name of a cmdlet, function, script file, or operable program. Check the spelling of the name, or if a path was included, verify that the path is correct and try again.'"

From the ISE, I can run get-command set-variable and can see that it is in Microsoft.PowerShell.Utility 3.1.0.0.  However, the command appears to only be available interactively which is a change with WMF 5.0.

**Comment by ShaunInIdaho** [comment]
Also note: this seems to only happen when my $ErrorActionPreference is set to 'Stop'.  I don't see any negative implications of letting the error silently continue.

**Comment by chshrikh** [comment]
@sameer-kumar we excuted you scrip with our environment ( OS - Server 2012 R2 and Win10 ), we are not able to repro the issue. 
1) Is it possible to share the detailed logs. You can mail us at RM_Customer_Queries at microsoft dot com
2) Just to confirm can you share the powershell version details . This can be done by executing command from powershell command prompt - $PSVersionTable 
3) Could you please let us know which OS you are using

**Comment by sameer-kumar** [comment]
Past week I was able to convince my team to use WMF 5.0 so no action needed at this point in time. Thanks for following-up on this.

**Comment by chshrikh** [comment]
Thank for update @sameer-kumar .
